### ISSUE TITLE
Auto-scroll in `console view` and rendering responses

### ISSUE THREAD

**Issue by DrorDvash** [its-a-feature/Mythic#391]
Hi there,
in “console view” there is no auto-scroll to see the output, so after pressing `enter` to send a command you have to move the hand and scroll to see the output every time.

![image](https://github.com/user-attachments/assets/f304d74b-1e21-4e8e-b8b8-8bbbf731c281)

One more “bug” I noticed - when you are doing `ls`, at first you get the raw output (without browser script - first screenshot), so if you want to get the “rendered” view (2nd screenshot) you need to click on **another task** and then click back on the `ls` task to see the formatted view. when you navigate in the filesystem of a victim and look for interesting files, it isn't very pleasant to do so every time (every `ls`)

raw output:
![image](https://github.com/user-attachments/assets/9b5b473f-ec0f-41c6-9726-a83457ac4758)

after click on another task and coming back to the task, the result is fine:
![image](https://github.com/user-attachments/assets/c886df77-08ac-4252-9195-c77b48f4cf2d)

in both cases, this was checked on macOS Sonoma 14.5 (m2), with poseidon agent.

**Comment by its-a-feature** [comment]
Once this latest build is done (https://github.com/its-a-feature/Mythic/actions/runs/10148225060) can you give that a shot? I added a bunch of UI updates to help with this and the auto scrolling issues. It's for Mythic 3.3 though.

**Comment by DrorDvash** [comment]
Yes i will! How to upgrade existing Mythic to  branch 3.3 (if its not in master) ?  I don't to mess things up so asking nooby question.
Thanks!

**Comment by its-a-feature** [comment]
No worries! I wouldn't recommend upgrades during an assessment, so if you're actively operating for something then I'd suggest holding off. Since Mythic3.3 is currently in beta as we're testing out some new features, the process is a _little_ more involved, but not much. Essentially:
* checkout Mythic 3.3
* sudo make (as usual) to get the new mythic-cli binary
* install the Mythic3.3 version of agents/c2 (ex: `sudo ./mythic-cli install github https://github.com/MythicAgents/poseidon -b Mythic3.3 -f`)
* repeat for each c2 and agent you're using (note: not all might have a Mythic3.3 beta branch currently,  https://mythicmeta.github.io/overview/)
* sudo ./mythic-cli start

**Comment by its-a-feature** [comment]
That work for you like you expect with the Mythic 3.3 updates?

**Comment by DrorDvash** [comment]
@its-a-feature 
I'm unsure what is wrong, but maybe you forgot to update the UI?
I did the steps twice but still not change. how can i be sure i'm on Mythic3.3?

![image](https://github.com/user-attachments/assets/99c55994-0478-4944-b5ae-f0b3ad936426)

**Comment by its-a-feature** [comment]
I believe the issue is here:
> ubuntu@ip-10-0-9-160:~/Mythic$ git checkout -b Mythic3.3
Switched to a new branch 'Mythic3.3'

I think what you did was update your branch (git pull), then create a new local branch called `Mythic3.3` (based on your current, Master branch), then you built the mythic-cli (from master). When you used `mythic-cli` to install the Mythic3.3 version of Poseidon though, that worked correctly, so you have Mythic3.2 and Poseidon for Mythic3.3 (hence the issue).

I think it might be as simple as like `git checkout -b Mythic3.3 origin/Mythic3.3` which says to switch to a local branch called Mythic3.3 which pulls changes from the origin/remote Mythic3.3 branch.

**Comment by DrorDvash** [comment]
> I believe the issue is here:
> 
> > ubuntu@ip-10-0-9-160:~/Mythic$ git checkout -b Mythic3.3
> > Switched to a new branch 'Mythic3.3'
> 
> I think what you did was update your branch (git pull), then create a new local branch called `Mythic3.3` (based on your current, Master branch), then you built the mythic-cli (from master). When you used `mythic-cli` to install the Mythic3.3 version of Poseidon though, that worked correctly, so you have Mythic3.2 and Poseidon for Mythic3.3 (hence the issue).
> 
> I think it might be as simple as like `git checkout -b Mythic3.3 origin/Mythic3.3` which says to switch to a local branch called Mythic3.3 which pulls changes from the origin/remote Mythic3.3 branch.

ok done.
now back to the original issue -> i don't see any auto-scroll.
typing a command (in this example `getuser`) will NOT scroll down and show the output when received, so it's the same as before.
![image](https://github.com/user-attachments/assets/0ad208c7-e5ae-4eb8-aa2b-d7d381cf7cd3)


This is how i expected it to be:

![image](https://github.com/user-attachments/assets/a77af982-6a66-4700-9633-229433a9f08f)

**Comment by its-a-feature** [comment]
Gotcha, I'll look into why the auto scroll seems to be working for the normal view but not the console view. You mentioned another issue originally about sometimes things like `ls` would show JSON until you clicked away and clicked back. Does that still happen for you?

**Comment by its-a-feature** [comment]
awesome! I'm glad that part is fixed! I think in the browserscripts there's definitely room to adjust the desired widths for certain columns to make sure they are wide enough to show all the data. That's an easy thing we can adjust per script (Mythic isn't defining it, each script is saying like "give me a table with these columns and here's each column's widths")

**Comment by its-a-feature** [comment]
ok, try pulling the latest from the Mythic3.3 branch https://github.com/its-a-feature/Mythic/actions/runs/10270124358 and seeing if that works for you now with the scroll. It should scroll to the output if you're at the bottom of the scroll window, but if you're scrolled up, then it shouldn't auto scroll down on you

**Comment by DrorDvash** [comment]
> ok, try pulling the latest from the Mythic3.3 branch https://github.com/its-a-feature/Mythic/actions/runs/10270124358 and seeing if that works for you now with the scroll. It should scroll to the output if you're at the bottom of the scroll window, but if you're scrolled up, then it shouldn't auto scroll down on you

I've probably missed your last reply. I have a few days off from work but be sure I'll test it out!
Will update here asap, thanks!

**Comment by DrorDvash** [comment]
It seems to work fine :) Thanks @its-a-feature 

regarding the errors while running `ps` command in poseidon, should it bother me? do I need to do something about it?
```
2024-08-27 02:24:15.607 pos[2753:32509] CFPropertyListCreateFromXMLData(): Old-style plist parser: missing semicolon in dictionary on line 3. Parsing will be abandoned. Break on _CFPropertyListMissingSemicolon to debug.
2024-08-27 02:24:15.621 pos[2753:32509] CFPropertyListCreateFromXMLData(): Old-style plist parser: missing semicolon in dictionary on line 3. Parsing will be abandoned. Break on _CFPropertyListMissingSemicolon to debug.
2024-08-27 02:24:15.634 pos[2753:32509] CFPropertyListCreateFromXMLData(): Old-style plist parser: missing semicolon in dictionary on line 3. Parsing will be abandoned. Break on _CFPropertyListMissingSemicolon to debug.
```

**Comment by its-a-feature** [comment]
Nah, you don't need to worry about them. The `poseidon` agent when doing `ps` used to try to do some parsing for the backing plist files for the processes, but macOS keeps changing things and has its own invalid data, so the parsing would throw errors. That's all been adjusted in the latest `poseidon` code

**Comment by its-a-feature** [comment]
I think we have all of these addressed now, so I'm going to close this issue. If you have other issues please continue to open up issues so I can get them resolved :)
